focus-ios/Blockzilla/Tracking Protection/TrackingProtectionPageStats.swift

```diff
@@ -192,28 +192,27 @@ private class TPStatsBlocklists {
             return nil
         }
 
-        domainSearch: for rule in rules {
-            // First, test the top-level filters to see if this URL might be blocked.
-            if rule.regex.firstMatch(in: resourceString, options: .anchored, range: resourceRange) != nil {
-                // Check the domain exceptions. If a domain exception matches, this filter does not apply.
-                for domainRegex in (rule.domainExceptions ?? []) {
-                    if domainRegex.firstMatch(in: resourceString, options: [], range: resourceRange) != nil {
-                        continue domainSearch
-                    }
+        for rule in rules {
+            guard let _ = rule.regex.firstMatch(in: resourceString, options: .anchored, range: resourceRange) else {
+                continue
+            }
+            let hasDomainException = rule.domainExceptions?.contains { domainRegex in
+                domainRegex.firstMatch(in: resourceString, options: [], range: resourceRange) != nil
+            } ?? false
+            guard !hasDomainException else {
```

Comment:
nit: Although guarding can make the early exits more readable in this case the inversion makes me a bit confused.
I would use `if hasDomainException { continue }` implying that if we indeed find an exception we skip to the next rule.